gobblin-data-management/src/main/java/org/apache/gobblin/util/commit/CreateDirectoryWithPermissionsCommitStep.java

```diff
@@ -49,9 +48,16 @@ public class CreateDirectoryWithPermissionsCommitStep implements CommitStep {
   private final URI fsUri;
```

Comment:
should javadoc give guidance to use this as a `PrePublishStep`, or could it be used either way?

...also is it worth have javadoc of each reference the other (e.g. if they're meant for companion use)?

Reply:
There's a scenario where `SetPermissionCommitStep` doesn't join together with `CreateDirectoryWithPermissionsCommitStep` is used for recursive copyable datasets, so I'll keep them separated for now. I think it's arguable to make this default behavior for all distcp modes but need to take precautions before enforcing this behavior.